### ISSUE TITLE
How does learning rate change during training stage

### ISSUE THREAD

**Issue by danieljf24** [fchollet/keras#1436]
I'd like to see How the learning rate change during the training stage, so I print the learning rate with `Callback`. The RMSprop optimizer is able to change the learning rate itself, but I find they not change with the following code.
Any help is greatly appreciated. Thanks a lot.

rms = RMSprop()
model.compile(loss='categorical_crossentropy', optimizer=rms)

class LearningRatePrinter(Callback):
    def __init__(self):
        super(LearningRatePrinter, self).__init__()

    def on_epoch_begin(self, epoch, logs={}):
        print 'lrearning_rate:', self.model.optimizer.lr.get_value()

lr_printer = LearningRatePrinter()
model.fit(X_train, Y_train, batch_size=batch_size, nb_epoch=nb_epoch, show_accuracy=True, verbose=2, callbacks=[lr_printer])

**Comment by ymcui** [comment]
Hi,
I don't think the learning rate changes, unless you set a `callback` to change learning rate yourself in your training process.
You'd better to see the equations of these optimizers, such as `rmsprop`, `adadelta` etc. 
They use other strategies for adapting the learning speed, but not change the so called `learning rate`.

**Comment by danieljf24** [comment]
thanks very much, ymcui.
